server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -0,0 +1,363 @@
+        stderrMsg.put("plugin_name", plugin.getClass().getSimpleName());
+        stderrMsg.put("plugin_version", plugin.version());
+
+        stderrMsg.put("input", !Strings.isNullOrEmpty(input) ? new JSONObject(input) : "");
+
+        Throwable pluginError = plugin.getPluginError();
+        JSONObject error = new JSONObject();
+        error.put("error", pluginError.getClass().getSimpleName());
+        error.put("msg", pluginError.getMessage());
+        error.put("stacktrace", pluginError.getStackTrace());
+
+        stderrMsg.put("err", error);
+
+        return stderrMsg.toString();
+    }
+
+    /**
+     * This is a synchronous commits and will block until either the commit succeeds
+     * or an unrecoverable error is encountered.
+     */
+    private void doCommitSync() {
+        try {
+            connection.commitSync();
+            logger.debug("Committed the processed record...");
+        } catch (WakeupException e) {
+            // we're shutting down, but finish the commit first and then
+            // rethrow the exception so that the main loop can exit
+            doCommitSync();
+            throw e;
+        } catch (CommitFailedException e) {
+            // the commit failed with an unrecoverable error. if there is any
+            // internal state which depended on the commit, you can clean it
+            // up here. otherwise it's reasonable to ignore the error and go on
+            logger.error("Commit failed", e);
+        }
+    }
+
+    /**
+     * Sets up a connection for producing error logs of a plug-in into a Kafka topic.
+     *
+     * @param serverAddress address of server
+     * @param clientID      client id
+     * @return properties for producer
+     */
+    private Properties setKafkaProducer(String serverAddress, String clientID) {
+
+        Properties p = new Properties();
+        p.setProperty(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, serverAddress);
+        p.setProperty(ProducerConfig.CLIENT_ID_CONFIG, clientID);
+        p.setProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+                StringSerializer.class.getName());
+        p.setProperty(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                StringSerializer.class.getName());
+
+        return p;
+    }
+
+    /**
+     * Get host name of this consumer.
+     *
+     * @return consumers host name
+     */
+    private String getConsumerHostName() {
+        try {
+            return InetAddress.getLocalHost().getHostName();
+        } catch (UnknownHostException e) {
+            logger.error("Could not find the consumer's hostname.");
+        }
+        return "Unknown";
+    }
+
+    /**
+     * This is a utility method to get current committed offset of all partitions for a consumer.
+     *
+     * @param consumer Kafka consumer
+     * @param topics   list of topics
+     */
+    private void getOffsetForPartitions(KafkaConsumer<String, String> consumer,
```

Comment:
This method looks unused in my IDE at least

Reply:
I removed it